apps/meteor/tests/unit/server/services/omnichannel/queue.tests.ts

```diff
@@ -0,0 +1,462 @@
+			settings.get.returns(true);
+			getConfig.returns({ autoAssignAgent: false });
+
+			const queue = new OmnichannelQueue();
+			queue.stop = Sinon.stub();
+			await queue.shouldStart();
+
+			expect(queue.stop.calledOnce).to.be.true;
+			expect(queue.stop.calledAfter(getConfig)).to.be.true;
+		});
+	});
+	describe('reconciliation', () => {
+		beforeEach(() => {
+			models.LivechatInquiry.removeByRoomId.resetHistory();
+			models.LivechatInquiry.takeInquiry.resetHistory();
+		});
+
+		it('should remove inquiries from rooms that do not exist', async () => {
+			const queue = new OmnichannelQueue();
+			await queue.reconciliation('missing', { roomId: 'rid', inquiryId: 'inquiryId' });
+
+			expect(models.LivechatInquiry.removeByRoomId.calledOnce).to.be.true;
+		});
+		it('should take an inquiry if the room was taken', async () => {
+			const queue = new OmnichannelQueue();
+			await queue.reconciliation('taken', { roomId: 'rid', inquiryId: 'inquiryId' });
+
+			expect(models.LivechatInquiry.takeInquiry.calledOnce).to.be.true;
+		});
+		it('should remove inquiries from rooms that were closed', async () => {
+			const queue = new OmnichannelQueue();
+			await queue.reconciliation('closed', { roomId: 'rid', inquiryId: 'inquiryId' });
+
+			expect(models.LivechatInquiry.removeByRoomId.calledOnce).to.be.true;
+		});
+		it('should return true for any other case', async () => {
+			const queue = new OmnichannelQueue();
+			expect(await queue.reconciliation('random', { roomId: 'rid', inquiryId: 'inquiryId' })).to.be.true;
+		});
+	});
+	describe('processWaitingQueue', () => {
+		let clock: any;
+		beforeEach(() => {
+			models.LivechatRooms.findOneById.reset();
+			models.LivechatInquiry.takeInquiry.resetHistory();
+			delegateInquiry.resetHistory();
+			queueLogger.debug.resetHistory();
+			clock = Sinon.useFakeTimers();
+		});
+		afterEach(() => {
+			clock.restore();
+		});
+		after(() => {
+			models.LivechatRooms.findOneById.reset();
+			models.LivechatInquiry.takeInquiry.reset();
+			delegateInquiry.reset();
+			queueLogger.debug.reset();
+		});
+
+		it('should process the public queue when department is undefined', async () => {
+			const queue = new OmnichannelQueue();
+
+			expect(await queue.processWaitingQueue(undefined, mockedInquiry)).to.be.true;
+			expect(queueLogger.debug.calledWith('Processing inquiry inquiryId from queue Public'));
+			expect(models.LivechatRooms.findOneById.calledOnce).to.be.true;
+		});
+		it('should call removeInquiry when findOneById returns null', async () => {
+			models.LivechatRooms.findOneById.returns(null);
+
+			const queue = new OmnichannelQueue();
+			expect(await queue.processWaitingQueue('department1', mockedInquiry)).to.be.true;
+			expect(
+				queueLogger.debug.calledWith({
+					msg: 'Room from inquiry missing. Removing inquiry',
+					roomId: 'rid',
+					inquiryId: 'inquiryId',
+					step: 'reconciliation',
+				}),
```

Comment:
I don't think we should check the log message here, we can check if `reconciliation()` has been called with the correct parameters instead.

Reply:
both are valid ways imho. but that check is not for assuring the function received the valid params, is for assuring that reconciliation got called at the right step.

Since missing & closed call the same func, the log is a good way to diff them.